### PR TITLE
ADBDEV-1994 changes for build of gpdb  using Makefile from gpAux

### PR DESCRIPTION
**Pull request by deart2k**
We want to use one way for tests and rpm build

### CODE REVIEW

File: gpAux/Makefile [hunk 1]
```diff
@@ -150,7 +151,8 @@ endif
 # Configure in thirdparty libraries from the ext/ directory
 CONFIG_INCLUDES=$(BLD_THIRDPARTY_INCLUDE_DIR)
 CONFIG_LIBRARIES=$(strip $(BLD_THIRDPARTY_LIB_DIR) $(BLD_CXX_LIB_DIR))
-CONFIGFLAGS+= --with-openssl --with-pam --with-ldap --with-includes="$(CONFIG_INCLUDES)" --with-libraries="$(CONFIG_LIBRARIES)"
+#CONFIGFLAGS+= --with-openssl --with-pam --with-ldap --with-includes="$(CONFIG_INCLUDES)" --with-libraries="$(CONFIG_LIBRARIES)"
+CONFIGFLAGS+=  --with-libxslt --with-openssl --with-ldap --with-includes="$(CONFIG_INCLUDES)" --with-libraries="$(CONFIG_LIBRARIES)"
```
Comment (Stolb27) [inline gpAux/Makefile]: @deart2k Why did you remove `--with-pam` option?

**Review by Stolb27 — COMMENTED** [review]
@deart2k Will we resolve the problem with DESTDIR or try to implement `make dist` build in the next release?

File: gpAux/Makefile (hunk 1 above)
Comment (deart2k) [reply gpAux/Makefile]: We've built our rpm without this options and our build image doesn't have dependencies for --with-pam at this moment so I removed this option. I'm going to return --with-pam.

**Comment by deart2k** [comment]
> @deart2k Will we resolve the problem with DESTDIR or try to implement `make dist` build in the next release?

I've disabled check "Found ‘${BUILDROOT}’ in installed files" (I think we need to fix it but we can build without fix) and we can build rpms by this way.

File: arenadata/Dockerfile [hunk 2]
```diff
@@ -20,7 +20,7 @@ RUN yum makecache && \
     yum -y install libevent-devel libuuid-devel libxml2-devel libyaml-devel libzstd-devel && \
     yum -y install openssl-devel pam-devel readline-devel snappy-devel libuv-devel && \
     yum -y install apache-ivy libicu perl-ExtUtils-Embed perl-Env perl-JSON && \
-    yum -y install perl-IPC-Run perl-Test-Base && \
+    yum -y install perl-IPC-Run perl-Test-Base libxslt-devel && \
```
Comment (darthunix) [inline arenadata/Dockerfile:23]: I didn't get from the discussion in the JIRA ticket - why do we need the XSLT library? It seems to be something about XML, but I've never seen this requirement for ORCA that uses XML for data serialization.

File: concourse/scripts/common.bash [hunk 3]
```diff
@@ -31,17 +31,26 @@ function os_version() {
 	fi
 }
 
+function os_platform() {
+	if [[ -f "/etc/altlinux-release" ]]; then
+		uname -m
```
Comment (darthunix) [inline concourse/scripts/common.bash]: 1. According to the open group, uname is [guaranteed](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/uname.html) to return only hardware name with `-m` option (i.e. information about [instruction set architecture - ISA](https://en.wikipedia.org/wiki/Instruction_set_architecture) - used to interact with CPU: a [list](https://stackoverflow.com/questions/45125516/possible-values-for-uname-m) of values in Linux). Processor architecture (`-p` - a microarchitecture for the target ISA) is a [Linux](https://man7.org/linux/man-pages/man1/uname.1.html)-specific thing (macOS also supports it, but Alt Linux returns `unknown`).
2. Anyway, it is incorrect to rely on processor microarchitecture, we should rather use ISA (CPU public instruction API) for this purpose.

Maybe we should better get rid of `uname -p` and rename all the result names? It seems to be a candidate for the upstream PR as well.

Comment (Stolb27) [reply concourse/scripts/common.bash]: I was confused by solaris uname [man page](https://docs.oracle.com/cd/E19109-01/tsolaris8/817-0879/6mgl9vnhr/index.html) that manifests:
>  -m Print the machine hardware name (class). **Use of this option is discouraged; use uname -p instead.** See NOTES section below.
-p Print the **current host's ISA** or processor type.

Though, of course, we may decrease patch size by replacing to `uname -m` because we can test this command on all supported platforms.

File: arenadata/Dockerfile (hunk 2 above)
Comment (Stolb27) [reply arenadata/Dockerfile:23]: This option isn't about ORCA at all. It enables SQL function for applying XSLT transformation to XML document. We have this option enabled by unknown historical reasons and have no statistics about usages of this value. It's the only cause to migrate this option from our CI to gpdb repo.

Comment (Stolb27) [reply arenadata/Dockerfile:23]: @darthunix I'm sorry, I provided an only partial answer. This library is required for `--with-libxslt`. I said about this option above.

File: concourse/scripts/common.bash (hunk 3 above)
Comment (Stolb27) [reply concourse/scripts/common.bash]: Solved in c73c627f017e8189535911b928390034d6c521b9

File: gpAux/releng/set_bld_arch.sh [hunk 4]
```diff
@@ -4,15 +4,15 @@
 
 case "`uname -s`" in
     Linux)
-    if [ -f /etc/redhat-release ]; then
+    if [ -z /etc/altlinux-release -a -f /etc/redhat-release ]; then
```
Comment (Stolb27) [inline gpAux/releng/set_bld_arch.sh:7]: @deart2k I've modified this script too (8e3ce288f22105367885aa2cdbc47e346140f387). Please, take a look.

Comment (deart2k) [reply gpAux/releng/set_bld_arch.sh:7]: All ok, got better

**Review by Stolb27 — APPROVED** [review]
